ui/pages/providers/[id].js

```diff
@@ -0,0 +1,167 @@
+  const [error, setError] = useState('')
+  const [errors, setErrors] = useState({})
+
+  useEffect(() => {
+    setName(provider?.name)
+  }, [provider])
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
+    setErrors({})
+    setError('')
+
+    try {
+      await mutate('/api/providers', async () => {
+        const res = await fetch(`/api/providers/${id}`, {
+          method: 'PATCH',
+          body: JSON.stringify({
+            id,
+            name,
+            clientSecret,
+          }),
+        })
+
+        const data = await res.json()
+
+        if (!res.ok) {
+          throw data
+        }
+
+        return {}
+      })
+    } catch (e) {
+      if (e.fieldErrors) {
+        const errors = {}
+        for (const error of e.fieldErrors) {
+          errors[error.fieldName.toLowerCase()] =
+            error.errors[0] || 'invalid value'
+        }
+        setErrors(errors)
+      } else {
+        setError(e.message)
+      }
+
+      return false
+    }
+
+    router.replace('/providers')
+
+    return false
+  }
+
+  return (
+    <div className='mx-auto w-full max-w-2xl'>
+      <Head>
+        <title>Edit Identity Provider</title>
+      </Head>
+      <header className='my-6 flex items-center justify-between'>
+        <h1 className='py-1 font-display text-xl font-medium'>Edit provider</h1>
+      </header>
+      <div className='flex w-full flex-col'>
+        <form onSubmit={onSubmit} className='mb-6 space-y-8'>
+          {/* Overview */}
+          <div>
+            <label className='text-2xs font-medium text-gray-700'>Name</label>
+            <input
+              type='text'
+              value={name}
+              onChange={e => {
+                setName(e.target.value)
+                setErrors({})
+                setError('')
+              }}
+              className={`mt-1 block w-full rounded-md shadow-sm focus:border-blue-500 focus:ring-blue-500 sm:text-sm ${
+                errors.name ? 'border-red-500' : 'border-gray-300'
+              }`}
+            />
+            {errors.name && <ErrorMessage message={errors.name} />}
+          </div>
+          <div className='w-full'>
+            <div className='mt-6 space-y-3'>
+              {provider?.kind !== 'google' && (
+                <div>
+                  <label className='text-2xs font-medium text-gray-700'>
+                    URL (Domain)
+                  </label>
+                  <input
+                    type='text'
+                    value={provider?.url}
+                    readOnly
+                    className={`mt-1 block w-full rounded-md border-gray-300 bg-gray-200 text-gray-600 shadow-sm focus:border-gray-300 focus:ring-0 sm:text-sm`}
+                  />
+                </div>
+              )}
+
+              <div>
+                <label className='text-2xs font-medium text-gray-700'>
+                  Client ID
+                </label>
+                <input
+                  readOnly
```

Comment:
Why are some fields editable and others not?

Reply:
We only want to support editing the name (because it's optional to set it, so user might want to nickname it later), 
and replacing the client secret (due to rotation)

URL and Domain should not be modified, but rather added as a new provider (and remove the current one if no longer needed)